lib/logstash/outputs/s3.rb

```diff
@@ -335,20 +383,29 @@ def shutdown_upload_workers
   end
 
   private
-  def handle_event(encoded_event)
+  def handle_event(encoded_event, event)
+    actualprefix = event.sprintf(@prefix)
+    if not @prefixes.to_a().include? actualprefix
```

Comment:
We prefer not using `not` in conditionals but instead `!`

``ruby
if !@prefixes ...
```